### PR TITLE
Add tdnf remove cache script and run it for marketplace images

### PR DESCRIPTION
**Pull request by mbykhovtsev-ms**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Adds a script that cleans and removes tdnf cache. Runs the script as post-install for marketplace generated images. Needed to fix bug https://microsoft.visualstudio.com/OS/_workitems/edit/45977096

###### Change Log  

- Adds a script that cleans and removes tdnf cache
- Runs tdnf remove cache as post-install step for marketplace image generation

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Built new ISO images locally, installed Mariner from ISO images and booted up. Confirmed that the tdnf cache was clear inside of installed Mariner VM

### CODE REVIEW

File: toolkit/imageconfigs/postinstallscripts/remove-tdnf-cache.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+echo removing tdnf cache
```
Comment (anphel31) [inline toolkit/imageconfigs/postinstallscripts/remove-tdnf-cache.sh:1]: Can you add the copyright notice to the top of this script?
Ex: https://github.com/microsoft/CBL-Mariner/blob/3d3fcf7ecb1300e436d07a59ed78279ac89f5985/.pipelines/prchecks/PackageBuildPRCheck.yml#L1C1-L2C34

**Comment by mbykhovtsev-ms** [comment]
> Maybe a future todo, but is there a reason we don't want to integrate this into our tools rather than use a customization script? This seems like a fairly standard thing we might want to do across all of our images.

This can be added to our tools for all images. I was just not sure if we needed them for all images or just the marketplace images, the bug I was trying to solve only talks about marketplace images, so I wanted to consult with the team whether it is needed for all of them? If it does, I can just integrated it as part of our `cleanup.sh` script that is run for all images

**Comment by dmcilvaney** [comment]
> > Maybe a future todo, but is there a reason we don't want to integrate this into our tools rather than use a customization script? This seems like a fairly standard thing we might want to do across all of our images.
> 
> This can be added to our tools for all images. I was just not sure if we needed them for all images or just the marketplace images, the bug I was trying to solve only talks about marketplace images, so I wanted to consult with the team whether it is needed for all of them? If it does, I can just integrate it as part of our `cleanup.sh` script that is run for all images

I think that is unique to just the container image. I was thinking more towards the imager tool itself, with a flag in the config schema. I'm not aware of any good reason to keep the cache in any image.

**Review by tobiasb-ms — APPROVED** [review]

File: toolkit/imageconfigs/marketplace-gen1-fips.json [hunk 2]
```diff
@@ -67,6 +67,9 @@
             "PostInstallScripts": [
                 {
                     "Path": "additionalconfigs/configure-image.sh"
+                },
+                {
+                    "Path": "postinstallscripts/remove-tdnf-cache.sh"
```
Comment (PawelWMS) [inline toolkit/imageconfigs/marketplace-gen1-fips.json:72]: The change looks fine but I'd consider these two modifications to either:
- add the contents of the new script into `configure-image.sh` or
- re-name `configure-image.sh` to something more precise like `configure-systemd-networkd.sh`.

Currently, I'm leaning more towards the second option, so the configuration scripts are more granular and their responsibility is clearer.

Comment (mbykhovtsev-ms) [reply toolkit/imageconfigs/marketplace-gen1-fips.json:72]: Added the renaming

**Pull request merged**